src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.cs

```diff
@@ -0,0 +1,304 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    /// <summary>
+    /// CA1835: Prefer Memory/ReadOnlyMemory overloads for Stream ReadAsync/WriteAsync methods.
+    ///
+    /// Undesired methods (available since .NET Framework 4.5):
+    ///
+    /// - Stream.WriteAsync(Byte[], Int32, Int32)
+    /// - Stream.WriteAsync(Byte[], Int32, Int32, CancellationToken)
+    /// - Stream.ReadAsync(Byte[], Int32, Int32)
+    /// - Stream.ReadAsync(Byte[], Int32, Int32, CancellationToken)
+    ///
+    /// Preferred methods (available since .NET Standard 2.1 and .NET Core 2.1):
+    ///
+    /// - Stream.WriteAsync(ReadOnlyMemory{Byte}, CancellationToken)
+    /// - Stream.ReadAsync(Memory{Byte}, CancellationToken)
+    ///
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class PreferStreamAsyncMemoryOverloads : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1835";
+
+        private static readonly LocalizableString s_localizableTitleRead = new LocalizableResourceString(
+            nameof(MicrosoftNetCoreAnalyzersResources.PreferStreamReadAsyncMemoryOverloadsTitle),
+            MicrosoftNetCoreAnalyzersResources.ResourceManager,
+            typeof(MicrosoftNetCoreAnalyzersResources));
+
+        private static readonly LocalizableString s_localizableMessageRead = new LocalizableResourceString(
```

Comment:
Read and Write seem to use the same message string.
```suggestion
        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(
```

Reply:
Ah thanks, good catch.